src/analyses/base.ml

```diff
@@ -1183,6 +1183,9 @@ struct
     not is_alloc || (is_alloc && not (ctx.ask (Queries.IsHeapVar v)))
 
   let query_invariant ctx context =
+    let keep_local = GobConfig.get_bool "ana.base.invariant.local" in
+    let keep_global = GobConfig.get_bool "ana.base.invariant.global" in
```

Comment:
Base is a functor that is dynamically instantiated after the configuration is loaded. One may consider extracting `keep_local` and `keep_global` out, such that these configurations do not have to be looked up everytime.

Reply:
It's possible, but I don't think it'll make a difference, base analysis uses so many other options probably a lot more often.